libs/state/docs/api/transformation-helpers/object/merge-objects.md

```diff
@@ -0,0 +1,38 @@
+# mergeObjects
+
+Merges object of type T with updates of type Partial T.
+Returns new object where updated overrides original values.
+Not mutating original object.
+
+_Example_
+
+```TypeScript
+interface Creature {
+ id: number,
+ type: string,
+ name: string
+}
+
+const cat = {id: 1, type: 'cat'};
```

Comment:
```suggestion
const cat = {id: 1, type: 'cat', name: 'Leo'};
```
From what I've seen about the type `function mergeObjects<T extends object>(object: T, upd: Partial<T>): T` one field is missing from object above. Here the `upd` input has a field that `object` input doesn't.

Reply:
Yes, nice. Also we can remove interface above because it is redundant.